tests/wallet/test_util.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+import re
+from random import shuffle
+from secrets import token_bytes
+from typing import Any, List
+
+import pytest
+from chia_rs import Coin
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_spend import CoinSpend
+from chia.types.condition_opcodes import ConditionOpcode
+from chia.util.ints import uint64
+from chia.wallet.util.compute_hints import (
+    HintedCoin,
+    hinted_coin_from_condition,
+    hinted_coin_with_coin_id,
+    hinted_coins_in_coin_spend,
+)
+
+
+def random_coin() -> Coin:
+    return Coin(bytes32(token_bytes(32)), token_bytes(32), uint64.from_bytes(token_bytes(8)))
```

Comment:
I think randomness in tests is a bad idea, even if the idea is that everything _should_ pass. We used to have randomness in the wallet tests and it resulted in so much flakiness that wasn't really bugs so much as bad compounding randomness.  Also if we decide to test a specific edge case, this makes it harder and likely has to be done anyways.

Reply:
Added 2c249db828f225680769f4e738ebf6769644c2fd.